teamcity: failed tests on master: testrace/TestHeartbeatHealth, test/TestHeartbeatHealth

The following tests appear to have failed:

[#771008](https://teamcity.cockroachdb.com/viewLog.html?buildId=771008):


```
--- FAIL: testrace/TestHeartbeatHealth (0.220s)
context_test.go:268: wanted ErrNotHeartbeated, not rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp: address notlocalserver: missing port in address"
--- FAIL: test/TestHeartbeatHealth (0.260s)
context_test.go:268: wanted ErrNotHeartbeated, not rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp: address notlocalserver: missing port in address"

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/rpc/context_test.go
+++ b/pkg/rpc/context_test.go
@@ -206,9 +206,24 @@ func TestHeartbeatHealth(t *testing.T) {
 		}
 	}()
 
+	lisNotLocalServer, err := net.Listen("tcp", "127.0.0.1:0")
+	defer func() {
+		netutil.FatalIfUnexpected(lisNotLocalServer.Close())
+	}()
+	if err != nil {
+		t.Fatal(err)
+	}
+	lisLocalServer, err := net.Listen("tcp", "127.0.0.1:0")
+	defer func() {
+		netutil.FatalIfUnexpected(lisLocalServer.Close())
+	}()
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	clientCtx := newTestContext(clock, stopper)
-	clientCtx.Addr = "notlocalserver"
-	clientCtx.AdvertiseAddr = "localserver"
+	clientCtx.Addr = lisNotLocalServer.Addr().String()
+	clientCtx.AdvertiseAddr = lisLocalServer.Addr().String()
 	// Make the interval shorter to speed up the test.
 	clientCtx.heartbeatInterval = 1 * time.Millisecond
 	if _, err := clientCtx.GRPCDial(remoteAddr).Connect(context.Background()); err != nil {
@@ -254,7 +269,14 @@ func TestHeartbeatHealth(t *testing.T) {
 		return clientCtx.ConnHealth(remoteAddr)
 	})
 
-	if err := clientCtx.ConnHealth("non-existent connection"); err != ErrNotHeartbeated {
+	lisNonExistentConnection, err := net.Listen("tcp", "127.0.0.1:0")
+	defer func() {
+		netutil.FatalIfUnexpected(lisNonExistentConnection.Close())
+	}()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := clientCtx.ConnHealth(lisNonExistentConnection.Addr().String()); err != ErrNotHeartbeated {
 		t.Errorf("wanted ErrNotHeartbeated, not %v", err)
 	}
 

